### PR TITLE
Add CI for netcoreappaot

### PR DESCRIPTION
**Pull request by morganbr**
Add CI for netcoreappaot on Windows. This only builds without running tests (similar to UAPAOT) since AOT testing is very slow.

### CODE REVIEW

**Comment by jkotas** [comment]
Is there anything extra that this covers that is not covered by the existing "Packaging All Configurations x64 Debug Build" leg?

**Comment by morganbr** [comment]
@ericstj would have a more authoritative answer, but I think there are a couple of differences (though neither is completely critical if we really don't want this leg; I added it based on @joperezr's suggestion that it would be better to catch issues in CI than an official build):
1. The All Configurations leg tries to build a minimal set of files. For example, since System.Console doesn't have a special netcoreappaot configuration, All Configurations will just build the netcoreapp flavor and reuse it for netcoreappaot. It's possible to make changes that only work for the individual TargetGroup or for All Configurations (usually by making incorrect assumptions about which TargetGroup gets used where, which I've done a few times). 
2. The All Configurations build only builds managed code. Currently, the native build is identical between netcoreapp and netcoreappaot though, so that's not very important.

**Comment by morganbr** [comment]
I should add that the existing All Configurations leg already did catch a netcoreappaot bug for @GrabYourPitchforks, so perhaps it's sufficient, depending on what other folks think of the coverage differences.

**Comment by morganbr** [comment]
@dotnet-bot test CI please

**Comment by morganbr** [comment]
@dotnet-bot test Windows NetCoreAppAOT x64 Release Build

**Comment by morganbr** [comment]
It just caught a real bug due to an incorrect configuration for the System.RegularExpressions reference assembly (I happen to already have a fix in #32045).

**Comment by ericstj** [comment]
> @ericstj would have a more authoritative answer

I actually think you covered it pretty well in https://github.com/dotnet/corefx/pull/32179#issuecomment-419610995.  AllConfigurations does its best with a minimal amount of overhead to capture all possible builds, but you can always go off the rails and do something that would break the official build.  In general folks don't make these types of changes a lot so I wouldn't be too worried about having that coverage.

I do see how its valueable for folks working on this to have coverage.  Maybe a middle ground is support for the leg, without having it on by default?  That way you can choose to run it on your PRs?

**Comment by morganbr** [comment]
Based on discussion with a few folks, we don't want this by default. @ericstj, do you know how to make it available on demand?

**Comment by ericstj** [comment]
I believe that is what this is doing:
https://github.com/dotnet/corefx/blob/da4ba45267002473d956a0c82624ad886fe55c54/buildpipeline/pipelinejobs.groovy#L49-L55

I think you can just leave `ForPR` as empty.

**Comment by morganbr** [comment]
@dotnet-bot test CI

**Comment by morganbr** [comment]
@dotnet-bot test CI

**Comment by wtgodbe** [comment]
@dotnet-bot test CI

File: buildpipeline/pipelinejobs.groovy [hunk 1]
```diff
@@ -31,6 +31,7 @@ def configurations = [
     ['TGroup':"netfx",      'Pipeline':winPipeline, 'Name':'NETFX', 'ForPR':"Release-x86"],
     ['TGroup':"uap",        'Pipeline':winPipeline, 'Name':'UWP CoreCLR', 'ForPR':"Debug-x64"],
     ['TGroup':"uapaot",     'Pipeline':winPipeline, 'Name':'UWP NETNative', 'ForPR':"Release-x86"],
```
Comment (wtgodbe) [inline buildpipeline/pipelinejobs.groovy]: nit: Could you fix indenting so these all match?

**Comment by wtgodbe** [comment]
@morganbr are you still planning on having this go in?

**Comment by morganbr** [comment]
I'll switch this to use WebAssembly for maximum coverage since the Windows version is mostly covered by -allconfigurations.

**Comment by stephentoub** [comment]
@morganbr, should this be closed? Or if not, will it be ready to go soon?

**Comment by stephentoub** [comment]
@morganbr, I'm going to close this for now, since it appears to be stale.  Obviously feel free to re-open if/when you're ready to continue with it.  Thanks!

**Pull request closed without merging**